app/src/main/java/com/twilio/video/app/ui/room/RoomActivity.java

```diff
@@ -770,68 +484,15 @@ private boolean permissionsGranted() {
                 && (resultStorage == PackageManager.PERMISSION_GRANTED));
     }
 
-    /** Initialize local media and provide stub participant for primary view. */
-    private void setupLocalMedia() {
-        localAudioTrack = LocalAudioTrack.create(this, true, MICROPHONE_TRACK_NAME);
-        setupLocalVideoTrack();
-        renderLocalParticipantStub();
-    }
-
-    /** Create local video track */
-    private void setupLocalVideoTrack() {
-
-        // initialize capturer only once if needed
-        if (cameraCapturer == null) {
-            cameraCapturer =
-                    new CameraCapturerCompat(this, CameraCapturer.CameraSource.FRONT_CAMERA);
-        }
-
-        cameraVideoTrack =
-                LocalVideoTrack.create(
-                        this,
-                        true,
-                        cameraCapturer.getVideoCapturer(),
-                        videoConstraints,
-                        CAMERA_TRACK_NAME);
-        if (cameraVideoTrack != null) {
-            localVideoTrackNames.put(
-                    cameraVideoTrack.getName(), getString(R.string.camera_video_track));
-
-            // Share camera video track if we are connected to room
-            if (localParticipant != null) {
-                localParticipant.publishTrack(cameraVideoTrack);
-            }
-        } else {
-            Snackbar.make(
-                            primaryVideoView,
-                            R.string.failed_to_add_camera_video_track,
-                            Snackbar.LENGTH_LONG)
-                    .show();
-        }
-    }
-
-    /**
-     * Render local video track.
-     *
-     * <p>NOTE: Stub participant is created in controller. Make sure to remove it when connected to
-     * room.
-     */
-    private void renderLocalParticipantStub() {
-        participantController.renderAsPrimary(
-                localParticipantSid,
-                getString(R.string.you),
-                cameraVideoTrack,
-                localAudioTrack == null,
-                cameraCapturer.getCameraSource() == CameraCapturer.CameraSource.FRONT_CAMERA);
-
-        primaryVideoView.showIdentityBadge(false);
+    private void updateUi(RoomViewState viewState) {
+        updateUi(viewState, false);
     }
 
-    private void updateUi(Room room) {
-        updateUi(room, false);
+    private void updateUi(boolean isAppLinkProvided) {
+        updateUi(null, isAppLinkProvided);
```

Comment:
I don't think we should allow a null RoomViewState to be passed to updateUi. We should be able to get an initial RoomViewState from the RoomManager even when the activity is launched as a result of a deep link click

Reply:
Good catch! I went ahead and removed this overloaded method. The RoomManager updateState function guards against null RoomViewState values and also the MutableLiveData viewState property initializes with a default RoomViewState instance.